Lecture-2/Task2/l2task2/Application.java

```diff
@@ -0,0 +1,79 @@
+package l2task2;
+
+import static java.lang.Math.random;
+
+public class Application {
+
+    private static Ticket[] createTickets(int amount) {
+        Ticket[] tickets = new Ticket[amount];
+        for (int i = 0; i < tickets.length; i++) {
+            tickets[i] = new Ticket(i + 1, "Theme #" + (i + 1));
+        }
+        return tickets;
+    }
+
+    private static Ticket[] distributeTickets(int studentsAmount, Ticket[] tickets) {
+        Ticket[] distributedTickets = new Ticket[studentsAmount];
+        for (int i = 0; i < distributedTickets.length; i++) {
+            distributedTickets[i] = tickets[(int) (random() * (tickets.length - 1) + 1)];
+        }
+        return distributedTickets;
+    }
+
+    private static int[] divideScores(int studentsAmount) {
+        int[] scores = new int[studentsAmount];
+        for (int i = 0; i < scores.length; i++) {
+            scores[i] = (int) (random() * 5 + 1);
+        }
+        return scores;
+    }
+
+    private static void showExamResults(Exam exam) {
+        System.out.println("Exam results of Group #" + exam.getGroup().getGroupId() + "->");
+        for (int i = 0; i < exam.getGroup().getStudents().length; i++) {
+            System.out.println(exam.getGroup().getStudents()[i] + " got ticket= "
+                    + exam.getTickets()[i] + " and score= " + exam.getScores()[i]);
+        }
+    }
+
+    private static void showMediumGroupScore(Exam exam) {
+        int sumOfAllScores = 0;
+        for (int score : exam.getScores()) {
+            sumOfAllScores += score;
+        }
+        System.out.println("Medium score of Group#" + exam.getGroup().getGroupId() + " is "
+                + (double) sumOfAllScores / exam.getScores().length);
+    }
+
+    private static int bestScoreInGroup(int[] scores) {
+        int bestScore = 1;
+        for (int score : scores) {
+            bestScore = Math.max(score, bestScore);
+        }
+        return bestScore;
+    }
+
+    private static void showStudentsByScores(int score, Exam exam) {
```

Comment:
It's still pretty difficult to figure out which student got which ticket. If somehow tickets ordering will be changed, all results will be lost
So, better approach is to create a class `ExamAnswer` with fields `Student, Ticket and Score`. `Exam` class in this case will hold references to `Group` and `List<ExamAnswer>`

Reply:
done